tslint.json

```diff
@@ -23,6 +23,42 @@
       "member-variable-declaration",
       "object-destructuring",
       "array-destructuring"
-    ]
+    ],
+    "whitespace": [
+      true,
+      "check-branch",
+      "check-decl",
+      "check-operator",
+      "check-module",
+      "check-separator",
+      "check-rest-spread",
+      "check-type",
+      "check-typecast",
+      "check-type-operator",
+      "check-preblock"
+    ],
+    "typedef-whitespace": [
+      true,
+      {
+        "call-signature": "nospace",
+        "index-signature": "nospace",
+        "parameter": "nospace",
+        "property-declaration": "nospace",
+        "variable-declaration": "nospace"
+      },
+      {
+        "call-signature": "onespace",
+        "index-signature": "onespace",
+        "parameter": "onespace",
+        "property-declaration": "onespace",
+        "variable-declaration": "onespace"
+      }
+    ],
+    "indent": [true, "spaces", 2],
+    "variable-name": [true, "allow-leading-underscore", "allow-pascal-case"],
+    "trailing-comma": false,
```

Comment:
What? Trailing commas are a good thing, aren't they? It helps a lot for cleaner diffs on PR.

Reply:
Firstly, this part had been taken from Icarus. 😛 Secondly, I've got used to trailing comma but initially, I'm not a big fan of it neither. Frankly, I've never seen how it could help us.
To make it clear, let's vote on this comment, put your thumb up/down to agree/disagree.
👍 - trailing commas must be in place, 👎 - don't check for trailing commas.

Reply:
From ESLint docs:

> Trailing commas simplify adding and removing items to objects and arrays, since only the lines you are modifying must be touched. Another argument in favor of trailing commas is that it improves the clarity of diffs when an item is added or removed from an object or array.
>
> Less clear:
>
> ```diff
>  var foo = {
> -    bar: "baz",
> -    qux: "quux"
> +    bar: "baz"
>  };
> ```
>
> More clear:
> 
> ```diff
>  var foo = {
>      bar: "baz",
> -    qux: "quux",
>  };
> ```

I vote for enforcing it 🙂

Reply:
All right, I see most people want to keep the trailing commas in place. Sorry Fahad, I'm gonna change this rule 😄